internal/ingress/controller/parser/parser.go

```diff
@@ -1740,26 +1785,69 @@ func toKongPlugin(plugin plugin) kong.Plugin {
 	return result
 }
 
-func kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) kong.Plugin {
-	return toKongPlugin(plugin{
+func (p *Parser) kongPluginFromK8SClusterPlugin(
+	k8sPlugin configurationv1.KongClusterPlugin) (kong.Plugin, error) {
+	var configError error
+	config := k8sPlugin.Config
+	if k8sPlugin.ConfigFrom !=
+		(configurationv1.NamespacedSecretValueFromSource{}) &&
+		len(k8sPlugin.Config) > 0 {
+		return kong.Plugin{},
+			errors.Errorf("KongClusterPlugin '/%v' has both "+
+				"Config and ConfigFrom set", k8sPlugin.Name)
+	}
+	if k8sPlugin.ConfigFrom != (configurationv1.
+		NamespacedSecretValueFromSource{}) {
+		config, configError = p.namespacedSecretToConfiguration(
```

Comment:
Much better than before. Please change `configError` to `err`. This error is not used anywhere and keeping it simply should help.